### PR TITLE
chore: update test_tools to read a file instead of get a password

### LINKED ISSUE #37: Swap out get_password test with read_file
We have an integration test which issues a `get_password` function. This is not a very good test because it doesn't use arguments, and I've seen in debug output LLMs passing args where they aren't needed. Also, the LLM can be sensitive to the command get_password, and I've at least once seen output asking confirmation because it knows such a command is dangerous:

```
{"role":"assistant","content":"You want me to retrieve your password to authenticate the current session. Just to confirm, I can only use that function with your consent and authentication. Would you like me to proceed?"}
```

I suggest we swap it out for something goose would use and is non-destructive. That way, the integration tests can pass or fail for the right reasons (e.g. not because it is worried about displaying a password). Also, they become realistic to the most prominent user of this library.

Here's an example, though we'd have to convert it to basic exchange abstraction for the test.
https://github.com/square/goose/blob/6fd11e8e4569621a57157e27d94cbd94a692eddf/src/goose/toolkit/developer.py#L117-L127

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Fixes #37 and avoids some LLMs being overly sensitive about giving you a password.

### CODE REVIEW

File: tests/test_integration.py [hunk 1]
```diff
@@ -34,26 +34,38 @@ def test_simple(provider, model):
...
+        system="You are a helpful assistant. Expect to need to read a file using read_file.",
+        tools=(Tool.from_function(read_file),),
     )
 
-    ex.add(Message.user("Can you authenticate this session by responding with the password"))
+    ex.add(Message.user(f"Can you read the contents of this file: {temp_file}"))
 
     response = ex.reply()
 
-    # It's possible this can be flakey, but in experience so far haven't seen it
```
Comment (codefromthecrypt) [inline tests/test_integration.py:55]: I want to say this, but might be TMI

```
    # On occasion, the LLM may forget it can do tool calls and fail. Retry in this case until
    # we have logic to handle this internally (retrying on text instead of tool choices).
```

**Review by baxen — APPROVED** [review]
Looks great!

File: tests/test_integration.py [hunk 2]
```diff
@@ -34,26 +34,38 @@ def test_simple(provider, model):
...
         provider=provider,
         model=model,
-        system="You are a helpful assistant. Expect to need to authenticate using get_password.",
-        tools=(Tool.from_function(get_password),),
+        system="You are a helpful assistant. Expect to need to read a file using read_file.",
+        tools=(Tool.from_function(read_file),),
     )
 
-    ex.add(Message.user("Can you authenticate this session by responding with the password"))
+    ex.add(Message.user(f"Can you read the contents of this file: {temp_file}"))
```
Comment (baxen) [inline tests/test_integration.py:64]: There might be some ambiguity in this around if it actually shares the ouptut again in its response, i'd maybe rephrase as "What are the contents of this file: "

Comment (baxen) [reply tests/test_integration.py:64]: Adding this now so we can include in upcoming release!
